src/Models/User.php

```diff
@@ -74,24 +75,54 @@ public function sendInvites(array $items): void
                 continue;
             }
 
-            /** @var Invite $invite */
-            $invite = $this->invites()->create([
-                'email' => $item['email'],
-                'role_id' => $item['role'],
-                'token' => Str::random(24),
-            ]);
-
-            Mail::to($invite->email)->send(new InviteUser($invite));
-
-            // show notification after sending invitation email to user
-            Notification::make()
-                ->success()
-                ->title('Invitation Sent')
-                ->body('An email invitation has been successfully sent to ' . $item['email'])
-                ->send();
+            // check if email address belong to any registered user
+            $user = User::where('email', $item['email'])->first();
+
+            // email address does not belong to any registered user
+            if (! $user) {
+                /** @var Invite $invite */
+                $invite = $this->invites()->create([
+                    'email' => $item['email'],
+                    'role_id' => $item['role'],
+                    'token' => Str::random(24),
+                ]);
+
+                Mail::to($invite->email)->send(new InviteUser($invite));
+
+                // show notification after sending invitation email to user
+                Notification::make()
+                    ->success()
+                    ->title('Invitation Sent')
+                    ->body('An email invitation has been successfully sent to ' . $item['email'])
+                    ->send();
+
+            } else {
+                // add role to user if user does not have this role yet
+                $role = Role::find($item['role']);
+
+                if ($user->roles->contains($role)) {
+                    // show notification
+                    Notification::make()
+                        ->success()
+                        ->title('User already has this role')
+                        ->body('User ' . $item['email'] . ' has ' . $role->name . ' role already')
+                        ->send();
+                } else {
+                    // add role to user
+                    // notification, email will be handled by ModelHasRole model event "created"
+                    $user->roles()->attach($role);
+                }
+            }
+
         }
     }
 
+    // define roles relationship to use custom class ModelHasRole, to capture model event when creating new model_has_roles record
+    public function roles(): BelongsToMany
+    {
+        return $this->belongsToMany(Role::class, 'model_has_roles', 'model_id')->using(ModelHasRole::class);
+    }
```

Comment:
I recommend extending the Spatie trait's relationship instead of completely replacing it. E.g: 

`return $this->spatieRoles()->using(ModelHasRole::class)`, and define `spatieRoles` as an alias for the Trait's method.

That way:

- it remains a morphToMany relationship,
- We keep the added functionality (in case there is any) of the Spatie-package defined relationship. 

It should work if you update the `ModelHasRole` to extend MorphPivot instead of Pivot.

That way, we can also remove the `creating` event callback to manually set the model_type on `ModelHasRole`.